evm_arithmetization/src/memory/memory_stark.rs

```diff
@@ -261,8 +258,21 @@ impl<F: RichField + Extendable<D>, const D: usize> MemoryStark<F, D> {
                     || trace_col_vecs[SEGMENT_FIRST_CHANGE][i].is_one()
                     || trace_col_vecs[VIRTUAL_FIRST_CHANGE][i].is_one())
             {
-                // `mem_after_filter = filter * address_changed * (1 - is_stale)`
-                trace_col_vecs[MEM_AFTER_FILTER][i] = F::ONE;
+                // `maybe_in_mem_after = filter * address_changed * (1 - is_stale)`
+                trace_col_vecs[MAYBE_IN_MEM_AFTER][i] = F::ONE;
+                // For efficiency, `mem_after_filter` is forcefully left to 0 if:
+                // - value is zero AND
+                // - segment is not preinitialized.
+                // Else, it's set to `maybe_in_mem_after`, which is 1 only in the current case.
```

Comment:
I get which info you aimed to carry, but the wording seems a bit weird to me, and may confuse a non-familiar reader. Why not just saying that we filter away memory values that are 0 in non-preinitialized segments? 

also:
> // Else, it's set to `maybe_in_mem_after`, which is 1 only in the current case.

is confusing too, as it seems to imply this could have other values than 0 or 1? But we constrain it to be binary.